Visualization/src/FCCEventHandler.cpp

```diff
@@ -0,0 +1,143 @@
+
+#include "DD4hep/Printout.h"
+#include "DD4hep/Objects.h"
+#include "DD4hep/Factories.h"
+
+#include "podio/EventStore.h"
+#include "podio/ROOTReader.h"
+
+#include "datamodel/TrackClusterCollection.h"
+#include "datamodel/TrackHitCollection.h"
+#include "datamodel/CaloClusterCollection.h"
+#include "datamodel/CaloHitCollection.h"
+#include "datamodel/GenVertexCollection.h"
+#include "datamodel/MCParticleCollection.h"
+
+#include "TSystem.h"
+#include "TGMsgBox.h"
+
+#include <stdexcept>
+#include <climits>
+
+#include "FCCEventHandler.h"
+
+using DD4hep::DDEveParticleActor;
+using DD4hep::DDEveHitActor;
+using DD4hep::DDEveHit;
+
+
+static void* _createFCCEventHandler(const char*) {
+  vis::FCCEventHandler* h = new vis::FCCEventHandler();
+  return h;
+}
+
+DECLARE_CONSTRUCTOR(DDEve_FCCEventHandler, _createFCCEventHandler)
+
+namespace vis {
+
+/// Standard constructor
+FCCEventHandler::FCCEventHandler() : EventHandler() {}
+
+/// Default destructor
+FCCEventHandler::~FCCEventHandler() {}
+
+/// Access the number of events on the current input data source (-1 if no data source connected)
+long FCCEventHandler::numEvents() const {
+  if (m_hasFile) {
+    return m_ROOTReader.getEntries();
+  }
+  return -1;
+}
+
+/// Access to the collection type by name
+DD4hep::EventHandler::CollectionType FCCEventHandler::collectionType(const std::string& cl) const {
+  if (cl == "caloClusters")
+    return CALO_HIT_COLLECTION;
+  else if (cl == "clusters")
+    return TRACKER_HIT_COLLECTION;
+  else if (cl == "allGenParticles")
+    return PARTICLE_COLLECTION;
+  else
+    return NO_COLLECTION;
+}
+
+template<typename T>
+void clusterEveConversion(T& clustercoll, DDEveHitActor& actor) {
+  int collsize = clustercoll->size();
+  actor.setSize(collsize);
+  for (const auto& cluster : *clustercoll) {
+    fcc::BareCluster core = cluster.Core();
```

Comment:
Actually, after the recent changes in the EDM I believe we'd like it to use the positioned hit, not the cluster